users/views.py

```diff
@@ -137,3 +140,70 @@ def post(self, request):
             return JsonResponse({'MESSAGE': 'KEY_ERROR'}, status=400)
         except TypeError:
             return JsonResponse({'MESSAGE': 'TYPE_ERROR'}, status=400)
+
+class FindUsernameView(View):
+
+    def post(self, request):
+        try:
+            data  = json.loads(request.body)
+            name  = data['name']
+            email = data['email']
+            
+            if not utils.validate_email(email):
+                return JsonResponse({'MESSAGE': 'INVALID_EMAIL'}, status=400)
+            
+            forgotten_user = User.objects.filter(name=name, email=email)
+            
+            if not forgotten_user.exists():
+                return JsonResponse({'MESSAGE': 'WRONG_USER'}, status=400)
+
+            forgotten_user = forgotten_user[0]
+            return JsonResponse({
+                "USERNAME"  : forgotten_user.username[:-3]+ "*" * 3,
+                "CREATED_AT": forgotten_user.created_at
+            }, status=200)
+        except json.decoder.JSONDecodeError:
+            return JsonResponse({'MESSAGE': 'JSON_DECODE_ERROR'}, status=400)
+        except KeyError:
+            return JsonResponse({'MESSAGE': 'KEY_ERROR'}, status=400)
+        except TypeError:
+            return JsonResponse({'MESSAGE': 'TYPE_ERROR'}, status=400)
+ 
+class MakeTemporaryPasswordView(View):
+
+    def post(self, request):
+        try:
+            data     = json.loads(request.body)
+            username = data['username']
+            name     = data['name']
+            email    = data['email']
+
+            if not utils.validate_username(username):
+                return JsonResponse({'MESSAGE': 'INVALID_USERNAME'}, status=400)
+
+            if not utils.validate_email(email):
+                return JsonResponse({'MESSAGE': 'INVALID_EMAIL'}, status=400)
+
+            forgotten_user = User.objects.filter(
+                username = username, 
+                name     = name, 
+                email    = email
+            )
+            
+            if not forgotten_user.exists():
+                return JsonResponse({'MESSAGE': 'WRONG_USER'}, status=400)
+
+            forgotten_user          = forgotten_user[0]
+            random_password         = token_urlsafe()[:16]
+            forgotten_user.password = bcrypt.hashpw(
+                random_password.encode('utf-8'), 
+                bcrypt.gensalt()
+            ).decode("utf-8")
+            forgotten_user.save()
```

Comment:
아래와 같이 수정해보시면 어떨까요?
`filter()`로 db 레코드를 조회할 때 database에 요청을 보내고 QuerySet을 반환받습니다. 이후에 `filter().exists()`로 method chaining할 때, 다시한번 db를 조회합니다. (hitting한다고 합니다.) 

2번 데이터베이스를 조회한다고 하면 아래와 같은 방법이 훨씬 효율적인 방법일 것입니다.
존재 여부부터 판단하고 존재하지 않으면 이후 db에 요청을 보내지 않겠죠?

```suggestion
            if User.objects.filter(username = username, name = name, email = email).exists():
                user = User.objects.get(username = username, name = name, email = email)
                random_password = token_urlsafe()[:16]
                user.password = bcrypt.hashpw(random_password.encode('utf-8'), bcrypt.gensalt()).decode("utf-8")
                user.save()
                return JsonResponse({'PASSWORD': random_password}, status=201)
            return JsonResponse({'MESSAGE': 'WRONG_USER'}, status=400)
```

Reply:
알려주셔서 고맙습니다 :D